### DISCUSSION TITLE
Why set npm_config_argv in process.env?

### DISCUSSION THREAD

**Discussion by kenrick95** [pnpm/pnpm#4153] [Q&A] [answered]
Hi,

I'm currently trying pnpm out in a large private monorepo. Because of the large number of packages, during a CI job, we have a script to choose which packages to install. So for example, if only package A & B are changed, we run `pnpm install --filter package-A --filter package-B`). This works well when the number of changed packages are small.

However, when we have large amount of packages specified in `--filter` (i.e. thousands) and when pnpm tries to spawn a new git process to fetch a dependency (`packages/git-fetcher/src/index.ts`' s`execGit`), `pnpm install` will throw this error:

```
 E2BIG  Command failed with E2BIG: /usr/bin/git clone https://gitlab+deploy-token-[REDACTED_TOKEN]@[REDACTED_PRIVATE_GIT_REPO].git /builds/[REDACTED_CI_PATH]/2/.ci-caches/pnpm-store/v3/tmp/_tmp_204_ac3c2ede530…
spawn E2BIG
```

Further investigation led me to find out that it is caused by this line:

https://github.com/pnpm/pnpm/blob/f8e031518ed954aec685f6c7e756f4dbd412ec72/packages/pnpm/src/main.ts#L84

In my scenario, `spawn` failed with ` E2BIG ` due to large amount of data stored in `process.env['npm_config_argv']`

So yeah, I'm wonder why that line is set?

I couldn't find any issue or PR that introduce that change, only a commit: https://github.com/pnpm/pnpm/commit/ffaaf70f63ec34a6915e16d963bbd4e0664d53eb#diff-efe6e8c43fe0c807601afdc100b68ed8512e84217a308446bca550f3c21e1632R147

p.s. iirc `npm_config_argv` in process.env has been removed in npm v7 too, as per https://github.com/npm/rfcs/blob/main/implemented/0021-reduce-lifecycle-script-environment.md

p.p.s. so for my original problem, I reckon the solution is to limit the number of `--filter` when I call `pnpm install` 😅  however i'm still curious on why `process.env['npm_config_argv']` is set

Thanks

**Comment by zkochan** [comment, marked as answer]
We set it because npm was setting it and some packages were not working without it. I did not know that npm removed it. We can also remove it in the next major version. Or we can add an option to remove it earlier.
